wolfcrypt/src/port/maxim/max3266x.c

```diff
@@ -0,0 +1,921 @@
+                    MXC_TPU_Cipher_Config(mode, MXC_TPU_CIPHER_AES256);
+                    status = MXC_TPU_Cipher_AES_Encrypt(in, iv, enc_key,
+                                MXC_TPU_CIPHER_AES256, mode, data_size, out);
+                    break;
+                default:
+                    wolfSSL_CryptHwMutexUnLock();
+                    return WC_HW_E;
+                break;
+        }
+        wolfSSL_CryptHwMutexUnLock();
+        if (status != E_SUCCESS){
+            return WC_HW_E;
+        }
+        return E_SUCCESS;
+    }
+
+    int wc_MXC_TPU_AesDecrypt(const char *in, const char *iv,
+                                const char *dec_key,
+                                MXC_TPU_MODE_TYPE mode,
+                                unsigned int data_size,
+                                char *out, unsigned int keySize)
+    {
+        int status;
+        status = wolfSSL_CryptHwMutexLock();
+        if (status != 0)
+            return status;
+        switch (keySize) {
+            case MXC_AES_KEY_128_LEN:
+                MXC_TPU_Cipher_Config(mode, MXC_TPU_CIPHER_AES128);
+                status = MXC_TPU_Cipher_AES_Decrypt(in, iv, dec_key,
+                            MXC_TPU_CIPHER_AES128, mode, data_size, out);
+                break;
+            case MXC_AES_KEY_192_LEN:
+                MXC_TPU_Cipher_Config(mode, MXC_TPU_CIPHER_AES192);
+                status = MXC_TPU_Cipher_AES_Decrypt(in, iv, dec_key,
+                            MXC_TPU_CIPHER_AES192, mode, data_size, out);
+                break;
+            case MXC_AES_KEY_256_LEN:
+                MXC_TPU_Cipher_Config(mode, MXC_TPU_CIPHER_AES256);
+                status = MXC_TPU_Cipher_AES_Decrypt(in, iv, dec_key,
+                            MXC_TPU_CIPHER_AES256, mode, data_size, out);
+                break;
+            default:
+                wolfSSL_CryptHwMutexUnLock();
+                return WC_HW_E;
+            break;
+        }
+
+        wolfSSL_CryptHwMutexUnLock();
+        if (status != E_SUCCESS){
+            return WC_HW_E;
+        }
+        return E_SUCCESS;
+    }
+
+
+#endif
+
+
+#if defined(MAX3266X_SHA)
+
+    int wc_MXC_TPU_SHA_Init(wc_MXC_Sha *hash)
+    {
+        if (hash == NULL) {
+            return BAD_FUNC_ARG;  // Appropriate error handling for null argument
+        }
+        hash->msg = NULL;
+        hash->used = 0;
+        hash->size = 0;
+        return EXIT_SUCCESS;
+    }
+
+    int wc_MXC_TPU_SHA_Update(wc_MXC_Sha *hash, const unsigned char* data,
+                                unsigned int size)
+    {
+        void *p;
+        if(size != (0 || NULL)){
+            if ((hash== NULL) || (data == NULL))
+                return BAD_FUNC_ARG;
+            if (hash->size < hash->used+size) {
+                if (hash->msg == NULL) {
+                    p = XMALLOC(hash->used+size, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+                } else {
+                    p = XREALLOC(hash->msg, hash->used+size, NULL,
```

Comment:
Please conditionalize XREALLOC based on WOLFSSL_NO_REALLOC. In the non-realloc case just malloc new variable, copy and free old.